# index and frameRate
rgbIndex: 2
thrIndex: 0
frameRate: 20
oldCamera: False 

# size of img
size_RGB: !!python/tuple [640,480]
size_RGB_720p: !!python/tuple [1280,720]
size_RGB_Calib: !!python/tuple [640,480]
size_RGB_Calib_output: !!python/tuple [800,600]
size_TRM: !!python/tuple [640,512]
size_TRM_output: !!python/tuple [640,512]
size_Dual: !!python/tuple [640,480]

# rostopic names
topic_RGB: "rgbImg"
topic_RGB_Calib: "calibImg"
topic_TRM: "thermalImg"
topic_Dual: "dualImg"
topic_yolo: "yoloImg"
topic_Bbox: "Bbox"
topic_Radar: "radarPub"
topic_RadarImg: "radarImg"
topic_DistImg: "DistImg"
topic_GPS: "GPSinfo"
topic_PredictPath: "predictPath"

# path of weights, cfg, names
path_RGB_weights: "./weight/RGB_final.weights"
path_RGB_cfg: "./cfg/obj.cfg"
path_RGB_names: "data/obj.names"

path_TRM_weights: "./weight/Thermal_final.weights"
path_TRM_cfg: "./cfg/obj.cfg"
path_TRM_names: "data/obj.names"

# distance from radar to rgb (m)
img2Radar_x: -0.1
img2Radar_y: 1.3
img2Radar_z: 1.6

# preference
textTime: 0.75

# camera info
r2c: [0., -1., 0., 0., 0., -1., 1., 0., 0.]
rtheta: 0

# before 2020/09/03, "oldCamera" == True 
# K: [ 1000.84389,    0.     ,  315.09341,
#        0.     ,  600.89312,  245.5822 ,
#        0.     ,    0.     ,    1.     ]

# 1th fisheye camera 640x480 before 2021
D: [-0.364865, 0.111417, -0.003288, 0.001529, 0.000000]
K: [ 443.84389,    0.     ,  300.09341,
       0.     ,  448.89312,  259.5822 ,
       0.     ,    0.     ,    1.     ]
R: [ 1.,  0.,  0.,
          0.,  1.,  0.,
          0.,  0.,  1.]

# 2th  fisheye camera 20210106
# D: [-0.390002, 0.108899, -0.003929, -0.009796, 0.000000]
# K: [ 454.90794,    0.     ,  322.94029,
#        0.     ,  459.41836,  247.9148 ,
#        0.     ,    0.     ,    1.     ]
# R: [ 1.,  0.,  0.,
#        0.,  1.,  0.,
#        0.,  0.,  1.]

# 3th  fisheye camera720p ~now
# D: [-0.513691, 0.092793, 0.110949, 0.007925, 0.000000]
# K: [ 856.70953,    0.     ,  588.80171,
#        0.     ,  760.19011,  144.3627 ,
#        0.     ,    0.     ,    1.     ]
# R: [ 1.,  0.,  0.,
#        0.,  1.,  0.,
#        0.,  0.,  1.]


# 0724
# pts_RGB: !!python/list [[149, 96], [239, 373], [418, 107], [404, 336], [403, 374], [526, 437], [702, 345]]
# pts_TRM: !!python/list [[ 80, 52], [165, 334], [352,  57], [337, 293], [337, 331] ,[460, 396], [634, 302]]

# 0826
# pts_RGB: !!python/list [[96, 107], [304, 205], [675, 141], [80, 358], [371, 448], [377, 374], [517, 191], [696, 442], [670, 326], [623, 134]]
# pts_TRM: !!python/list [[ 31, 64], [238, 154], [618,  91], [11, 317], [303, 407], [311, 330], [455, 140], [632, 393], [604, 278], [560,  83]]
# ROI: !!python/tuple [[55,548],[73,697]]

# 0903 back3 
# pts_RGB: !!python/list [[331,183],[164,244],[598,267],[425,293],[49,279],[535,352]]
# pts_TRM: !!python/list [[323,215],[187,262],[545,289],[402,308],[89,289],[498,360]]
# ROI: !!python/tuple [[80,559],[80,719]]

# 0909
# pts_RGB: !!python/list [[164, 245], [610, 187], [286, 298], [331, 183], [145, 428], [165,  66], [312, 364], [104, 386]]
# pts_TRM: !!python/list [[188, 263], [555, 222], [283, 307], [323, 214], [165, 415], [186, 114], [305, 362], [138, 379]]

# 0924 Fisheye
# pts_RGB: !!python/list [[1009, 308], [993, 789], [817, 344], [1144, 676], [603, 714], [921, 479], [466, 395]]
# pts_TRM: !!python/list [[ 500,  37], [488, 373], [350,  48], [ 615, 295], [159, 319], [432, 156], [ 55,  88]]

# 1016 Fisheye rain19
# pts_RGB: !!python/list [[554,250],[450, 137],[548, 482],[412, 176],[396, 300],[366, 450],[407, 234],[257, 310],[307, 374]]
# pts_TRM: !!python/list [[515,173],[350,  12],[509, 499],[289,  73],[270, 246],[216, 469],[278, 149],[ 19, 251],[104, 370]]
# ROI: !!python/tuple [[154,467],[250,587]]

# 1026 Yung7
# pts_RGB: !!python/list [[273,290],[178,132],[252,315],[558,146],[463,208],[391,478],[173,403]]
# pts_TRM: !!python/list [[171,255],[37,3],[136,259],[626,33],[494,109],[379,508],[19,386]]
# ROI: !!python/tuple [[141,475],[164,550]]

# 1203 Costco1
pts_RGB: !!python/list [[245,212],[291,278],[468,244],[376,349],[227,435],[543,410]]
pts_TRM: !!python/list [[129,118],[180,201],[471,158],[325,306],[76,435],[592,395]]
ROI: !!python/tuple [[134,480],[186,567]]
